File: pdlearning/src/microservices/course-microservice/Api/Application/Models/AttendanceRatioOfPresentInfoModel.cs
using System;

namespace Microservice.Course.Application.Models
{
    public class AttendanceRatioOfPresentInfo
    {
        public Guid RegistrationId { get; set; }

        public int TotalSessions { get; set; }

        public int PresentSessions { get; set; }
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/course/models/attendance-ratio-of-present-info.model.ts
export interface IAttendanceRatioOfPresentInfo {
  registrationId: string;
  totalSessions: number;
  presentSessions: number;
}

export class AttendanceRatioOfPresentInfo implements IAttendanceRatioOfPresentInfo {
  public registrationId: string;
  public totalSessions: number;
  public presentSessions: number;

  constructor(data?: IAttendanceRatioOfPresentInfo) {
    if (data) {
      this.registrationId = data.registrationId;
      this.totalSessions = data.totalSessions;
      this.presentSessions = data.presentSessions;
    }
  }

  public displayAsText(): string {
    return `${this.presentSessions}/${this.totalSessions}`;
  }
}
